File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
//         instantiate map
       HashMap<Integer, Integer> map = new HashMap<>();
//         loop over nums:
        for (int i:nums) {
//             if key exist in map
            if (map.containsKey(i)) {
//                 increment its value
                map.put(i, map.get(i) + 1);
            } else {
//                 else, instantiate it
                map.put(i, 1);
            }
        }
        
//      instantiate variable for key and one for its value:
        int k = 0;
        int v = 0;
        for (int i:map.keySet()) {
            if (map.get(i) > v) {
                v = map.get(i);
                k = i;
            }
        }
        return k;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
#         instantiate a dictionary:
        dict = {}
#         loop over nums:
        for i in nums:
#         if it exists in dict, increment, else instantiate it
            if i in dict:
                dict[i] = dict[i] + 1
            else :
                dict[i] = 1
#                 set up k for result and v to hold the value
        k = 0;
        v = 0;
        for key in dict:
#             if the value of current key is bigger than v
            if dict[key] > v:
#         v will hold the new biggest value
                v = dict[key]
#     the new result is the current element or key
                k = key
#     return final key that holds greatest value
        return k
        